## 動作確認プログラム

#### 確認1　transformersインストール後の動作確認

Jax, TensorFlow, PyTorchをpipでインストールした後に行う。

In [1]:
# transformersインストール後の動作確認コード

from transformers import pipeline
classifier = pipeline('sentiment-analysis') 
results = classifier(["We are very happy to show you the 🤗 Transformers library.", 
           "We hope you don't hate it."]) 
for result in results: 
    print(f"label: {result['label']}, with score: {round(result['score'], 4)}")

/opt/conda/lib/python3.7/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
The cache for model files in Transformers v4.22.0 has been updated. Migrating your old cache. This is a one-time only operation. You can interrupt this and resume the migration later on by calling `transformers.utils.move_cache()`.


Moving 0 files to the new cache system


0it [00:00, ?it/s]
No model was supplied, defaulted to distilbert-base-uncased-finetuned-sst-2-english and revision af0f99b (https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english).
Using a pipeline without specifying a model name and revision in production is not recommended.
Downloading: 100%|██████████| 629/629 [00:00<00:00, 485kB/s]
Downloading: 100%|██████████| 268M/268M [00:04<00:00, 58.1MB/s] 
Downloading: 100%|██████████| 48.0/48.0 [00:00<00:00, 36.0kB/s]
Downloading: 100%|██████████| 232k/232k [00:00<00:00, 932kB/s] 


label: POSITIVE, with score: 0.9998
label: NEGATIVE, with score: 0.5309


label: POSITIVE, with score: 0.9998 

label: NEGATIVE, with score: 0.5309

インストールに問題なければ上記の2行が結果として表示される。

#### 確認2　Fugashiとその辞書（ipadic）のインストール後の動作確認

FugashiとFugashiから利用する辞書（ipadic）をpipでインストール

In [2]:
from transformers import pipeline
from transformers import AutoModelForSequenceClassification
from transformers import BertJapaneseTokenizer

TARGET_TEXT = "誰でもできる感情分析です。簡単なので、気軽に試してみましょう。"

classifier_model_name = 'daigo/bert-base-japanese-sentiment'
tokenizer_model_name = 'cl-tohoku/bert-base-japanese-whole-word-masking'

model = AutoModelForSequenceClassification.from_pretrained(classifier_model_name)
tokenizer = BertJapaneseTokenizer.from_pretrained(tokenizer_model_name)
nlp = pipeline("sentiment-analysis", model = model, tokenizer = tokenizer)

print(nlp(TARGET_TEXT))

Downloading: 100%|██████████| 1.32k/1.32k [00:00<00:00, 919kB/s]
Downloading: 100%|██████████| 443M/443M [00:07<00:00, 60.9MB/s] 
Downloading: 100%|██████████| 258k/258k [00:00<00:00, 1.41MB/s]
Downloading: 100%|██████████| 479/479 [00:00<00:00, 328kB/s]


[{'label': 'ポジティブ', 'score': 0.9763936400413513}]


問題なければ上記では以下と同様の結果が表示される。

[{'label': 'ポジティブ', 'score': 0.9750074148178101}]